src/electron/ipc/ipc-message-receiver.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Interpreter } from 'background/interpreter';
+import { Logger } from 'common/logging/logger';
+import { InterpreterMessage } from 'common/message';
+import { IpcRenderer, IpcRendererEvent } from 'electron';
+import { isObject, isString } from 'lodash';
+import { IPC_MESSAGE_CHANNEL_NAME } from './ipc-channel-names';
+
+export class IpcMessageReceiver {
+    constructor(
+        private readonly interpreter: Interpreter,
+        private readonly underlyingIpcRenderer: IpcRenderer,
+        private readonly logger: Logger,
+    ) {}
+
+    public initialize(): void {
+        this.underlyingIpcRenderer.on(IPC_MESSAGE_CHANNEL_NAME, this.onMessageReceived);
+    }
+
+    private onMessageReceived = (event: IpcRendererEvent, ...args: any[]) => {
+        if (event.senderId !== 0) {
```

Comment:
I'm curious: when does this happen?
This log here remind me of the log we added on AI-Web when we receive a message for a tab we are not tracking, that log became a spammer right away.

Reply:
Per electron docs, senderId is:

> The webContents.id that sent the message, you can call `event.sender.sendTo(event.senderId, ...)` to reply to the message, see `ipcRenderer.sendTo` for more information. This only applies to messages sent from a different renderer. Messages sent directly from the main process set `event.senderId` to 0.

That is, we don't currently expect it in any circumstance, since we don't have other processes sending messages besides the main process

Reply:
(updated the error message to make this more clear)